File: androidp1/app/src/main/java/com/preappointment1/app/data/api/ApiService.kt
package com.preappointment1.app.data.api

import com.preappointment1.app.data.model.*
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PATCH
import retrofit2.http.DELETE
import retrofit2.http.Path

interface ApiService {
    @POST("api/v1/auth/register")
    suspend fun register(@Body request: AuthRequest): AuthResponse

    @POST("api/v1/auth/login")
    suspend fun login(@Body request: AuthRequest): AuthResponse

    @POST("api/v1/auth/challenge")
    suspend fun deviceChallenge(@Body request: DeviceChallengeRequest): DeviceChallengeResponse

    @POST("api/v1/auth/device/register")
    suspend fun deviceRegister(@Body request: DeviceRegisterRequest): DeviceAuthResponse

    @POST("api/v1/auth/device/verify")
    suspend fun deviceVerify(@Body request: DeviceVerifyRequest): DeviceAuthResponse

    @POST("api/v1/auth/device/recover")
    suspend fun deviceRecover(@Body request: DeviceRegisterRequest): DeviceAuthResponse

    @POST("api/v1/auth/refresh")
    suspend fun refreshTokens(@Body request: RefreshTokenRequest): RefreshTokenResponse

    @POST("api/v1/agents/recommend")
    suspend fun recommendAgent(@Body request: RecommendRequest): AgentResponse

    @GET("api/v1/agents")
    suspend fun getAgents(): List<AgentResponse>

    @GET("api/v1/profiles")
    suspend fun getProfiles(): List<ProfileResponse>

    @POST("api/v1/profiles")
    suspend fun createProfile(@Body request: ProfileRequest): ProfileResponse

    @GET("api/v1/subscriptions")
    suspend fun getSubscriptions(): List<SubscriptionResponse>

    @POST("api/v1/subscriptions")
    suspend fun createSubscription(@Body request: SubscriptionRequest): SubscriptionResponse

    @PATCH("api/v1/subscriptions/{id}")
    suspend fun patchSubscription(
        @Path("id") id: String,
        @Body request: UpdateSubscriptionRequest
    ): SubscriptionResponse

    @DELETE("api/v1/subscriptions/{id}")
    suspend fun deleteSubscription(@Path("id") id: String)

    @DELETE("api/v1/auth/delete")
    suspend fun deleteAccount()

    @GET("api/v1/subscriptions/{id}/timeline")
    suspend fun getTimeline(@Path("id") subscriptionId: String): List<TimelineEventResponse>

    @POST("api/v1/subscriptions/{id}/timeline")
    suspend fun postTimelineEvent(
        @Path("id") subscriptionId: String,
        @Body request: TimelineEventRequest
    ): TimelineEventResponse

    @DELETE("api/v1/subscriptions/{id}/timeline/{eventId}")
    suspend fun deleteTimelineEvent(
        @Path("id") subscriptionId: String,
        @Path("eventId") eventId: String
    )
}

File: iosp1/P1/P1/Network/ApiService.swift
import Foundation

class ApiService {
    static let shared = ApiService()
    
    private let client = ApiClient.shared
    
    private init() {}
    
    // MARK: - Auth
    func register(request: AuthRequest, allowAuthRetry: Bool = true) async throws -> AuthResponse {
        return try await client.performRequest(
            path: "api/v1/auth/register",
            method: "POST",
            body: request,
            allowAuthRetry: allowAuthRetry
        )
    }

    func login(request: AuthRequest, allowAuthRetry: Bool = true) async throws -> AuthResponse {
        return try await client.performRequest(
            path: "api/v1/auth/login",
            method: "POST",
            body: request,
            allowAuthRetry: allowAuthRetry
        )
    }
    
    func deleteAccount() async throws {
        try await client.performRequest(path: "api/v1/auth/delete", method: "DELETE")
    }
    
    // MARK: - Agents
    func recommendAgent(request: RecommendRequest) async throws -> AgentResponse {
        return try await client.performRequest(path: "api/v1/agents/recommend", method: "POST", body: request)
    }
    
    func getAgents() async throws -> [AgentResponse] {
        return try await client.performRequest(path: "api/v1/agents", method: "GET")
    }
    
    // MARK: - Profiles
    func getProfiles() async throws -> [ProfileResponse] {
        return try await client.performRequest(path: "api/v1/profiles", method: "GET")
    }
    
    func createProfile(request: ProfileRequest) async throws -> ProfileResponse {
        return try await client.performRequest(path: "api/v1/profiles", method: "POST", body: request)
    }
    
    // MARK: - Subscriptions
    func getSubscriptions() async throws -> [SubscriptionResponse] {
        return try await client.performRequest(path: "api/v1/subscriptions", method: "GET")
    }
    
    func createSubscription(request: SubscriptionRequest) async throws -> SubscriptionResponse {
        return try await client.performRequest(path: "api/v1/subscriptions", method: "POST", body: request)
    }
    
    func patchSubscription(id: String, request: UpdateSubscriptionRequest) async throws -> SubscriptionResponse {
        return try await client.performRequest(path: "api/v1/subscriptions/\(id)", method: "PATCH", body: request)
    }
    
    func deleteSubscription(id: String) async throws {
        try await client.performRequest(path: "api/v1/subscriptions/\(id)", method: "DELETE")
    }
    
    // MARK: - Timeline
    func getTimeline(subscriptionId: String) async throws -> [TimelineEventResponse] {
        return try await client.performRequest(path: "api/v1/subscriptions/\(subscriptionId)/timeline", method: "GET")
    }
    
    func postTimelineEvent(subscriptionId: String, request: TimelineEventRequest) async throws -> TimelineEventResponse {
        return try await client.performRequest(path: "api/v1/subscriptions/\(subscriptionId)/timeline", method: "POST", body: request)
    }
    
    func deleteTimelineEvent(subscriptionId: String, eventId: String) async throws {
        try await client.performRequest(path: "api/v1/subscriptions/\(subscriptionId)/timeline/\(eventId)", method: "DELETE")
    }
}
